controllers/openstackdataplane_controller.go

```diff
@@ -205,12 +205,21 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 			err.Error()))
 		return ctrl.Result{}, err
 	}
+	r.Log.Info("found roles", "total", len(roles.Items))
+	if len(roles.Items) > 0 && len(instance.Spec.Roles) > len(roles.Items) {
+		shouldRequeue = true
+	}
 	if shouldRequeue {
 		r.Log.Info("one or more roles aren't ready, requeueing")
 		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
 	}
 	if instance.Spec.DeployStrategy.Deploy && len(deployErrors) == 0 {
 		instance.Status.Deployed = true
+		if len(instance.Spec.Roles) > 0 && len(roles.Items) == 0 {
```

Comment:
I think we don't need this here and why would you set `instance.Status.Deployed = true` a line before when `len(instance.Spec.Roles) >0`.  Would not removing condition `len(roles.Items) > 0` from L209 would requeue for the roles to be created if `len(roles.Items) == 0`.

Reply:
last time I tried to requeue when `len(roles.Items) == 0` it didn't reconcile `OpenstackDataplaneRole`, it kept requeuing `OpenstackDataplane`.
Let me test again to confirm

Reply:
Actually, the infinite requeue was not cause by this. I was running the debugger with webhooks disabled and the node wasn't getting the SSH secret 🤦🏽‍♂️